### PR TITLE
hourly forcast

### PR DESCRIPTION
**Pull request by shahriar113**

### CODE REVIEW

File: src/App.tsx [hunk 1]
```diff
@@ -1,13 +1,13 @@
...
-import HomePage from './pages/HomePage';
 import Cities from './pages/Cities';
+import HomePage from './pages/HomePage';
 import Map from './pages/Map';
 import Settings from './pages/Settings';
 
 function App() {
   return (
-    <div className="App h-screen w-screen">
+    <div className="App w-screen">
```
Comment (minaahilimtiaz) [inline src/App.tsx]: Why are your removing this property?

File: src/App.tsx [hunk 2]
```diff
@@ -1,13 +1,13 @@
 import { Route, BrowserRouter as Router, Routes } from 'react-router-dom';
 import './App.css';
-import HomePage from './pages/HomePage';
 import Cities from './pages/Cities';
+import HomePage from './pages/HomePage';
```
Comment (minaahilimtiaz) [inline src/App.tsx:4]: Why are you re-arranging the imports?

File: src/components/BaseComponent.tsx [hunk 3]
```diff
@@ -11,7 +11,7 @@ const BaseComponent: React.FC<BaseComponentProps> = ({
 }) => {
   return (
     <div
-      className={`bg-primary h-screen w-screen p-8 grid grid-cols-6 gap-4 ${baseComponentStyles}`}
+      className={`bg-primary w-screen p-8 grid grid-cols-6 gap-4 ${baseComponentStyles}`}
```
Comment (minaahilimtiaz) [inline src/components/BaseComponent.tsx]: What is the purpose of this change?

File: src/components/ThreeHourlyForecast.tsx [hunk 4]
```diff
@@ -1,11 +1,64 @@
-import React from 'react';
+import moment from 'moment';
+import React, { memo, useCallback, useEffect, useState } from 'react';
+import { commonHttp } from '../utils/commonHttp';
+import { WeatherIcon } from './WeatherIcon';
+
+// const url = 'https://api.openweathermap.org/data/2.5/forecast?lat=-23.8041&lon=90.4152&exclude=hourly&units=metric&appid=b462849fee8752d40630390beee9e87d'
```
Comment (minaahilimtiaz) [inline src/components/ThreeHourlyForecast.tsx]: remove this comment

File: src/components/ThreeHourlyForecast.tsx [hunk 5]
```diff
@@ -1,11 +1,64 @@
...
+              <div
+                className={`flex ${
+                  i !== data.length - 1
+                    ? ' border-r border-r-[#9399A2] px-4 border-opacity-50 '
+                    : ' px-4 border-opacity-50 '
+                }`}
+                key={i}
+              >
+                <div>
+                  <span className="text-[#9399A2]">
```
Comment (minaahilimtiaz) [inline src/components/ThreeHourlyForecast.tsx]: assign this hash value a colour then use it.

File: src/components/ThreeHourlyForecast.tsx [hunk 6]
```diff
@@ -1,11 +1,64 @@
...
+                    {moment(el?.dt_txt).format('LT')}
+                  </span>
+                  <div className="flex flex-col justify-center items-center mt-4 h-[70px] w-[70px]">
+                    {el?.weather?.length > 0 && (
```
Comment (minaahilimtiaz) [inline src/components/ThreeHourlyForecast.tsx:48]: don't pass hardcoded values like this.

File: src/components/WeatherIcon.tsx [hunk 7]
```diff
@@ -0,0 +1,48 @@
+import rainImage from '../assets/rain.svg'; // Import the image
+import sunImage from '../assets/sun-svgrepo-com.svg'; // Import the image
+import weatherImage from '../assets/weather.svg'; // Import the image
+import windImage from '../assets/wind.svg'; // Import the image
+
+export const WeatherIcon = ({ icon }: any) => {
+  // Define a mapping of icon codes to corresponding icon URLs
+  const iconMapping: any = {
+    // '01d': 'sunny.png',  // Replace with the actual URL or icon you want to use
```
Comment (minaahilimtiaz) [inline src/components/WeatherIcon.tsx]: remove the comments

File: src/pages/HomePage.tsx [hunk 8]
```diff
@@ -1,9 +1,9 @@
 import React from 'react';
-import WeeklyForecast from '../components/WeeklyForecast';
-import ThreeHourlyForecast from '../components/ThreeHourlyForecast';
+import BaseComponent from '../components/BaseComponent';
```
Comment (minaahilimtiaz) [inline src/pages/HomePage.tsx:2]: No need to rearrange the imports

**Review by minaahilimtiaz — APPROVED** [review]

File: src/App.tsx (hunk 1 above)
Comment (shahriar113) [reply src/App.tsx]: it was breaking design

File: src/components/BaseComponent.tsx (hunk 3 above)
Comment (shahriar113) [reply src/components/BaseComponent.tsx]: it was creating fixed height for background

File: src/App.tsx (hunk 1 above)
Comment (minaahilimtiaz) [reply src/App.tsx]: how?

File: src/components/BaseComponent.tsx (hunk 3 above)
Comment (minaahilimtiaz) [reply src/components/BaseComponent.tsx]: what do you mean?

File: src/App.tsx (hunk 2 above)
Comment (shahriar113) [reply src/App.tsx:4]: it is auto rearraning from my editor

File: src/App.tsx (hunk 1 above)
Comment (minaahilimtiaz) [reply src/App.tsx]: revert this change

File: src/components/BaseComponent.tsx (hunk 3 above)
Comment (minaahilimtiaz) [reply src/components/BaseComponent.tsx]: revert this change

File: src/components/WeeklyForecast.tsx [hunk 9]
```diff
@@ -3,7 +3,9 @@ import React from 'react';
 const WeeklyForecast: React.FC = () => {
   return (
     <div className="h-full w-full">
-      <h2 className="text-lg font-semibold">Add 7 day forecast here</h2>
```
Comment (minaahilimtiaz) [inline src/components/WeeklyForecast.tsx:6]: Revert it

File: src/components/ThreeHourlyForecast.tsx [hunk 10]
```diff
@@ -1,11 +1,63 @@
...
+                    ? ' border-r border_r_color_9399A2 px-4 border-opacity-50 '
+                    : ' px-4 border-opacity-50 '
+                }`}
+                key={i}
+              >
+                <div>
+                  <span className="text_9399A2">
+                    {moment(el?.dt_txt).format('LT')}
+                  </span>
+                  <div className="flex flex-col justify-center items-center mt-4 wh70">
```
Comment (minaahilimtiaz) [inline src/components/ThreeHourlyForecast.tsx]: please remove this wh70 property. it has disturbed whole layout

**Review by minaahilimtiaz — CHANGES_REQUESTED** [review]

File: src/components/ThreeHourlyForecast.tsx [hunk 11]
```diff
@@ -1,11 +1,63 @@
...
+
+  useEffect(() => {
+    getData();
+    return () => {};
+  }, []);
+
   return (
-    <div className="w-full bg-indigo-500 shadow-md rounded-md">
-      Add 3 hourly forecast
+    <div className="w-full bg-[#202b3b] shadow-md rounded-md py-4 px-5 whitespace-nowrap">
```
Comment (minaahilimtiaz) [inline src/components/ThreeHourlyForecast.tsx]: what is the purpose of this whitespace property?

**Review by minaahilimtiaz — CHANGES_REQUESTED** [review]

File: src/pages/HomePage.tsx [hunk 12]
```diff
@@ -15,7 +15,7 @@ const HomePage: React.FC = () => {
           <AirConditions />
         </div>
 
-        <div className="bg-green-950 shadow-md rounded-md col-span-2">
+        <div className="bg-[#202b3b] shadow-md rounded-md col-span-2 weeklyArea">
```
Comment (minaahilimtiaz) [inline src/pages/HomePage.tsx:18]: Remove this property weeklyArea property

**Review by minaahilimtiaz — CHANGES_REQUESTED** [review]
